src/controllers/krate/publish.rs

```diff
@@ -222,9 +222,10 @@ pub async fn publish(app: AppState, req: BytesRequest) -> AppResult<Json<GoodCra
             return Err(cargo_err("expected at most 5 categories per crate"));
         }
 
-        let max_features = existing_crate
-            .and_then(|c| c.max_features.map(|mf| mf as usize))
-            .unwrap_or(app.config.max_features);
+        let max_features = match &existing_crate {
+            Some(c) => c.max_features.map(|mf| mf as usize),
+            None => None,
+        }.unwrap_or(app.config.max_features);
```

Comment:
```suggestion
        let max_features = existing_crate.as_ref()
            .and_then(|c| c.max_features.map(|mf| mf as usize))
            .unwrap_or(app.config.max_features);
```

`.as_ref()` seems a bit easier to resolve the problem :)

Reply:
Ugh, pick the very last thing I did on a day when I was still recovering from being sick. :see_no_evil: